sycl/source/detail/pi.cpp

```diff
@@ -197,7 +197,12 @@ bool trace(TraceLevel Level) {
 
 // Initializes all available Plugins.
 vector_class<plugin> initialize() {
-  vector_class<plugin> Plugins;
+  static bool PluginsInitDone = false;
+  static vector_class<plugin> Plugins;
+  if (PluginsInitDone) {
+    return Plugins;
+  }
```

Comment:
It seems it is possible for two threads to reach this point, so both accesses `Plugins` var. This will lead to problems.

Reply:
Why? Plugins are really read-only immutable objects after they are constructed in pi::initilize(). We should probably make it explicitly such, so that people aren't able to write it.

Reply:
If two threads create platforms for the first time simultaneously they can reach this point or I missing something?